guides/micronaut-server-filter-request/groovy/src/main/groovy/example/micronaut/LoggingHeadersFilter.groovy

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package example.micronaut
+
+import io.micronaut.core.order.Ordered
+import io.micronaut.http.HttpHeaders
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.annotation.RequestFilter
+import io.micronaut.http.annotation.ServerFilter
+import io.micronaut.http.filter.ServerFilterPhase
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import static io.micronaut.http.annotation.Filter.MATCH_ALL_PATTERN
+
+@ServerFilter(MATCH_ALL_PATTERN) // <1>
+class LoggingHeadersFilter implements Ordered {
+    private static final Logger LOG = LoggerFactory.getLogger(LoggingHeadersFilter.class)
+
+    @RequestFilter // <2>
+    void filterRequest(HttpRequest<?> request) {
+        if (LOG.isTraceEnabled()) {
+            HttpHeaders headers = request.getHeaders()
+            for (String headerName : headers.names()) {
+                if (headerName.equalsIgnoreCase(HttpHeaders.AUTHORIZATION)) {
```

Comment:
not sure we should be encouraging unsafe patterns like this, how do you know what other headers might contain sensitive values?

Reply:
Would you prefer if I remove the part where we skip logging the Authorization header and instead log every HTTP header in the guide?

I think showing users to be careful about what they log is better than logging everything.

Please note that [in core, we skip logging some headers](https://github.com/micronaut-projects/micronaut-core/blob/4.3.x/http/src/main/java/io/micronaut/http/util/HttpHeadersUtil.java#L37-L38).  Of course, what header is sensitive is application specific.

Reply:
perhaps you could use `HeaderUtils` since it already does some masking

Reply:
[I have changed the Guide to use `HttpHeaderUtils`](https://github.com/micronaut-projects/micronaut-guides/pull/1408/commits/9fd2688186443555041bffafd3407392739e2e10)